tests/phpunit/src/TestBase.php

```diff
@@ -201,15 +201,15 @@ protected function setUp($output = NULL): void {
     $this->setClientProphecies();
     $this->setIo($input, $output);
 
-    $this->fixtureDir = realpath(__DIR__ . '/../../fixtures');
+    $this->fixtureDir = $this->getTempDir();
+    $this->fs->mirror(realpath(__DIR__ . '/../../fixtures'), $this->fixtureDir);
```

Comment:
This is where the magic happens, mirroring the fixture to a temporary directory rather than modifying it in-situ so there's no chance of cross-talk between tests